### PR TITLE
Stable Vaults External Rewards Function

### PR DESCRIPTION
**Pull request by cybertronprime**

### CODE REVIEW

File: proto/comdex/rewards/v1beta1/tx.proto [hunk 1]
```diff
@@ -14,6 +14,7 @@ service Msg {
   rpc ExternalRewardsLockers(ActivateExternalRewardsLockers) returns (ActivateExternalRewardsLockersResponse);
   rpc ExternalRewardsVault(ActivateExternalRewardsVault) returns (ActivateExternalRewardsVaultResponse);
   rpc ExternalRewardsLend(ActivateExternalRewardsLend) returns (ActivateExternalRewardsLendResponse);
+  rpc ExternalRewardsStableMint(ActivateExternalRewardsStableMint) returns (ActivateExternalRewardsStableMintResponse);
```
Comment (dheerajkd30) [inline proto/comdex/rewards/v1beta1/tx.proto:17]: This is not registered in codec.go, is it no required ?

Comment (cgsingh33) [reply proto/comdex/rewards/v1beta1/tx.proto:17]: added

File: x/vault/keeper/vault.go [hunk 2]
```diff
@@ -683,3 +683,129 @@ func (k Keeper) GetAmountOfOtherToken(ctx sdk.Context, id1 uint64, rate1 sdk.Dec
...
+				break
+			} else if userRewards.Amount.LT(quanitity) {
+				quanitity = quanitity.Sub(userRewards.Amount)
+				k.DeleteStableMintVaultRewards(ctx, userRewards)
+			}
+		}
+	}
+}
+
+func (k Keeper) GetStableMintVaultRewardsOfAllApps(ctx sdk.Context) (mappingData []types.StableMintVaultRewards) {
```
Comment (dheerajkd30) [inline x/vault/keeper/vault.go:789]: unused function . Any reason to introduce it ?

**Comment by cybertronprime** [comment]
Was added and used, but changed the functionality a bit, keeping it for
future reference

**Review by dheerajkd30 — APPROVED** [review]

**Review by pratikasr — APPROVED** [review]

**Pull request merged**